pipelines/_ingredients/ApPipe.yaml

```diff
@@ -103,6 +103,8 @@ subsets:
       An alias of ApPipe to use in higher-level pipelines.
 contracts:
   - detectAndMeasure.doSkySources == transformDiaSrcCat.doRemoveSkySources
+  # Do not run forced measurement if the diaForcedSources are not loaded.
+  - loadDiaCatalogs.doLoadForcedSources == diaPipe.doRunForcedMeasurement
```

Comment:
Strictly speaking it's fine if `loadDiaCatalogs.doLoadForcedSources` is set but `diaPipe.doRunForcedMeasurement` is not -- do you prefer the stricter condition?

Reply:
Please do add a contract for `apdb_config_url`, just in case someone or something bypasses the pipeline parameter.

Reply:
My thought for the forced sources was that it was wasteful to load them in `loadDiaCatalogs` if we don't run forced photometry in `diaPipe`. On thinking the logic through more carefully now, the preloaded forced sources are *only* used to package with the alerts, while the forced sources created with `diaPipe.doRunForcedMeasurement` are also written to the APDB. So we could potentially turn loading forced sources off to save time, but still generate new forced sources for the APDB. It is less useful to preload forced sources and not run forced photometry, but I could imagine cases where we would want to do that. Both scenarios are slightly unlikely I think, so I am inclined to either leave the strict condition on the assumption that we would turn both off if we turn off either, or remove the contract entirely.